src/phoenix/experimental/evals/functions/binary.py

```diff
@@ -100,24 +115,30 @@ def run_relevance_eval(
     ].map(list)
 
     exploded_df = filtered_df.explode(retrieved_documents_column_name, ignore_index=False)
-    exploded_df[retrieved_document_text_column_name] = [
-        document_data["document.content"] if document_data is not None else None
-        for document_data in exploded_df[retrieved_documents_column_name]
-    ]
+    if trace_data:
+        exploded_df[retrieved_document_text_column_name] = [
+            document_data["document.content"] if document_data is not None else None
+            for document_data in exploded_df[retrieved_documents_column_name]
+        ]
+    else:
+        exploded_df[retrieved_document_text_column_name] = [
+            document_data if document_data is not None else None
+            for document_data in exploded_df[retrieved_documents_column_name]
+        ]
```

Comment:
this needs to be extracted out of this function so that it can be made composable. Doing this imperatively like so won't scale. I'd think we extract this to make it composable and could optionally detect if the documents are just chunks or document objects - no need to force the parameterization.

Reply:
In both cases here we are dealing with Chunks just at different hierarchies in the objects based on trace data or not. 

Maybe the suggestion below of detecting trace_data is the right thing to do, prior to this section.  Might be a way to slightly clean it up based on that.

Reply:
Yeah I like the idea of structure detection - cc @fjcasti1